app/services/hyrax/file_set_characterizer.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Hyrax
+  ##
+  # Characterizes an ActiveFedora based FileSet
+  class FileSetCharacterizer
+    ##
+    # @!attribute [rw] source
+    #   @return [#characterize]
+    attr_accessor :source
+
+    ##
+    # @param source the object to characterize
+    def initialize(source:)
+      self.source = source
+    end
+
+    ##
+    # @return [void]
+    #
+    # @raise [RuntimeError] if FileSet is missing the characterization_proxy
+    def characterize
+      Hydra::Works::CharacterizationService.run(characterization_proxy, filepath)
+      Rails.logger.debug "Ran characterization on #{characterization_proxy.id} (#{characterization_proxy.mime_type})"
+      characterization_proxy.alpha_channels = channels(filepath) if source.image? && Hyrax.config.iiif_image_server?
+      characterization_proxy.save!
+      update_source
+      CreateDerivativesJob.perform_later(source, source.original_file.id, filepath)
+    end
+
+    private
+
+      def update_source
```

Comment:
I'm not clear on this method. It's updating the source and updating the parent's collections, but it's not updating the characterization proxy. I might be in favor of either putting this code back in `#characterize` until a clearer abstraction is found.